Frameworks/Foundation/NSDictionary.mm

```diff
@@ -825,35 +801,43 @@ - (NSArray*)objectsForKeys:(NSArray*)keys notFoundMarker:(id)notFoundMarker {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSSet*)keysOfEntriesPassingTest:(BOOL (^)(id, id, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self keysOfEntriesWithOptions:0 passingTest:predicate];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (NSSet*)keysOfEntriesWithOptions:(NSEnumerationOptions)opts passingTest:(BOOL (^)(id, id, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSSet*)keysOfEntriesWithOptions:(NSEnumerationOptions)options passingTest:(BOOL (^)(id, id, BOOL*))predicate {
+    __block NSMutableSet* ret = [NSMutableSet setWithCapacity:[self count]];
+    [self enumerateKeysAndObjectsWithOptions:options
+                                  usingBlock:^(id key, id obj, BOOL* stop) {
+                                      if (predicate(key, obj, stop)) {
+                                          [ret addObject:key];
+                                      }
+                                  }];
+
+    return ret;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Caveat
+ @Notes atomically parameter not supported
 */
-- (BOOL)writeToURL:(NSURL*)aURL atomically:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (BOOL)writeToURL:(NSURL*)url atomically:(BOOL)atomically {
+    NSError* error = nil;
+    NSData* data = [NSPropertyListSerialization dataWithPropertyList:self format:NSPropertyListXMLFormat_v1_0 options:0 error:&error];
+    if (error != nil) {
+        return NO;
+    }
+
+    return [data writeToURL:url atomically:atomically];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (BOOL)fileExtensionHidden {
```

Comment:
Whoops looks like this got changed cross branches. I'll revert this to mitigate merge conflicts w/ NSFileManager PR